finite_difference/src/argument_mod.f90

```diff
@@ -21,14 +21,19 @@ module argument_mod
    enumerator :: GO_MIN, GO_MAX, GO_SUM
 end enum
 
-! Fake array to enable us to provide stencil meta-data in the args() array
-integer, public, parameter :: stencil(0:111,0:111,0:111) = 0
+type :: go_stencil
+   integer :: first_row
+   integer :: second_row
+   integer :: third_row
+end type go_stencil
+
+public go_stencil
 
   !args(fs,stencil,arg_intent) ! this need defining
 type :: go_arg
-  integer :: arg_intent
+  integer(kind(GO_READ)) :: arg_intent
```

Comment:
I don't think we should use `GO_READ` as a kind parameter. If a kind is needed here then we should add a suitable parameter to kind_params_mod.f90 and use that instead.